tests/dialects/test_tsql.py

```diff
@@ -243,3 +243,28 @@ def test_convert_date_format(self):
                 "spark": "CAST(x AS INT)",
             },
         )
+
+    def test_add_date(self):
+        self.validate_identity("SELECT DATEADD(year, 1, '2017/08/25')")
+        self.validate_all(
+            "SELECT DATEADD(year, 1, '2017/08/25')", write={"spark": "SELECT ADD_MONTHS('2017/08/25', 12)"}
+        )
+        self.validate_all("SELECT DATEADD(qq, 1, '2017/08/25')", write={"spark": "SELECT ADD_MONTHS('2017/08/25', 3)"})
+        self.validate_all("SELECT DATEADD(wk, 1, '2017/08/25')", write={"spark": "SELECT DATE_ADD('2017/08/25', 7)"})
+
+    def test_date_diff(self):
+        self.validate_identity("SELECT DATEDIFF(year, '2021/01/01', '2020/01/01')")
+        self.validate_all(
+            "SELECT DATEDIFF(year, '2021/01/01', '2020/01/01')",
+            write={
+                "tsql": "SELECT DATEDIFF(year, '2021/01/01', '2020/01/01')",
+                "spark": "SELECT DATEDIFF('2020/01/01', '2021/01/01') / 12",
```

Comment:
Are you sure this is correct? Why do we divide by 12? Shouldn't the end date come first? Check [this](https://spark.apache.org/docs/2.3.0/api/sql/index.html#datediff).

Reply:
SELECT DATEDIFF(year, '2011/08/25', '2017/08/25') AS DateDiff; => 6
SELECT DATEDIFF('2011/08/25', '2017/08/25') => -1292

Where I am wrong and where the 12 confusion comes from is that this should convert into months_between